mongodb-native/src/main/java/org/ballerinalang/mongodb/MongoDBCollectionUtil.java

```diff
@@ -86,16 +109,25 @@ public static Object find(HandleValue collection, Object filter, Object sort, lo
             sort = EMPTY_JSON;
         }
         Document sortDoc = Document.parse(sort.toString());
-
-        MongoCollection<Document> mongoCollection = (MongoCollection<Document>) collection.getValue();
-        MongoCursor<Document> results;
         try {
+            MongoDatabase mongoDatabase = Utils.getCurrentDatabase(env, client, databaseName);
+            MongoCollection<Document> mongoCollection = mongoDatabase.getCollection(collectionName.getValue());
+            MongoCursor<Document> results;
             if (limit != -1) {
```

Comment:
Shall we add a description to find method in Ballerina Client explaining if limit=-1, the method will return all the results?